apps/stripe/src/app/api/saleor/transaction-process-session/use-case-response.ts

```diff
@@ -1,124 +1,25 @@
-      message: "Payment intent succeeded",
-    });
-
-    return Response.json(typeSafeResponse, { status: this.statusCode });
-  }
-}
-
-class AuthorizationSuccess extends ChargeSuccess {
-  readonly result: ResponseResult = "AUTHORIZATION_SUCCESS";
-}
-
-class ChargeActionRequired extends SuccessWebhookResponse {
-  readonly result: ResponseResult = "CHARGE_ACTION_REQUIRED";
-  readonly saleorMoney: SaleorMoney;
-  readonly stripeStatus: StripePaymentIntentStatus;
-  readonly stripePaymentIntentId: StripePaymentIntentId;
-
-  constructor(args: {
-    saleorMoney: SaleorMoney;
-    stripePaymentIntentId: StripePaymentIntentId;
-    stripeStatus: StripePaymentIntentStatus;
-  }) {
-    super();
-    this.stripeStatus = args.stripeStatus;
-    this.saleorMoney = args.saleorMoney;
-    this.stripePaymentIntentId = args.stripePaymentIntentId;
-  }
 
-  private getMessageFromStripeStatus() {
-    switch (this.stripeStatus) {
-      case "requires_action":
-        return "Payment intent requires action";
-      case "requires_confirmation":
-        return "Payment intent requires confirmation";
-      case "requires_payment_method":
-        return "Payment intent requires payment method";
-    }
-  }
-
-  getResponse() {
-    const typeSafeResponse = buildSyncWebhookResponsePayload<"TRANSACTION_PROCESS_SESSION">({
-      result: this.result,
-      amount: this.saleorMoney.amount,
-      pspReference: this.stripePaymentIntentId,
-      // https://docs.stripe.com/payments/paymentintents/lifecycle
-      message: this.getMessageFromStripeStatus(),
-    });
-
-    return Response.json(typeSafeResponse, { status: this.statusCode });
-  }
-}
-
-class AuthorizationActionRequired extends ChargeActionRequired {
-  readonly result: ResponseResult = "AUTHORIZATION_ACTION_REQUIRED";
-}
-
-class ChargeRequest extends SuccessWebhookResponse {
-  readonly result: ResponseResult = "CHARGE_REQUEST";
-  readonly saleorMoney: SaleorMoney;
-  readonly stripePaymentIntentId: StripePaymentIntentId;
-
-  constructor(args: { saleorMoney: SaleorMoney; stripePaymentIntentId: StripePaymentIntentId }) {
-    super();
-    this.saleorMoney = args.saleorMoney;
-    this.stripePaymentIntentId = args.stripePaymentIntentId;
-  }
-
-  getResponse() {
-    const typeSafeResponse = buildSyncWebhookResponsePayload<"TRANSACTION_PROCESS_SESSION">({
-      result: this.result,
-      amount: this.saleorMoney.amount,
-      pspReference: this.stripePaymentIntentId,
-      message: "Payment intent is processing",
-    });
-
-    return Response.json(typeSafeResponse, { status: this.statusCode });
-  }
-}
-
-class AuthorizationRequest extends ChargeRequest {
-  readonly result: ResponseResult = "AUTHORIZATION_REQUEST";
-}
-
-class ChargeFailure extends SuccessWebhookResponse {
-  readonly result: ResponseResult = "CHARGE_FAILURE";
+abstract class FailureBase extends SuccessWebhookResponse {
+  readonly result = "" as "CHARGE_FAILURE" | "AUTHORIZATION_FAILURE";
```

Comment:
you don't have to do that,  I believe base class is to share common props. You don't have to define it in base class, you can just do `abstract result: "CHARGE_FAILURE" | "AUTHORIZATION_FAILURE"` and it must be implemented later